runtime-extension-model/src/test/java/org/mule/runtime/core/extension/CoreExtensionDeclarerTestCase.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2023 Salesforce, Inc. All rights reserved.
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.extension;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.hasSize;
+import static java.util.stream.Collectors.toList;
+
+import io.qameta.allure.Issue;
+import org.mule.metadata.api.model.MetadataType;
+import org.mule.runtime.api.meta.model.declaration.fluent.ComponentDeclaration;
+import org.mule.runtime.api.meta.model.declaration.fluent.NamedDeclaration;
+import org.mule.runtime.api.meta.model.declaration.fluent.ParameterizedDeclaration;
+import org.mule.runtime.api.meta.model.declaration.fluent.ParameterizedDeclarer;
+import org.mule.runtime.api.metadata.resolving.AttributesTypeResolver;
+import org.mule.runtime.api.metadata.resolving.InputTypeResolver;
+import org.mule.runtime.api.metadata.resolving.OutputTypeResolver;
+import org.mule.runtime.api.metadata.resolving.TypeKeysResolver;
+import org.mule.runtime.core.api.extension.provider.MuleExtensionModelDeclarer;
+import org.mule.runtime.extension.api.metadata.ComponentMetadataConfigurer;
+import org.mule.sdk.api.metadata.resolving.ChainInputTypeResolver;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Issue("W-15261626")
+public class CoreExtensionDeclarerTestCase {
+
+  private List<ParameterizedDeclarer> routersOneOf = new ArrayList<>();
+  private List<ParameterizedDeclarer> routersAllOf = new ArrayList<>();
+  private List<ParameterizedDeclarer> scopesPassThrough = new ArrayList<>();
+
+  @Test
+  public void configurationOfExtensionDeclarer() {
+    MuleExtensionModelDeclarer declarer = new MuleExtensionModelDeclarer(MockConfigurer::new);
+    declarer.createExtensionModel();
+    assertThat(routersAllOf, hasSize(1));
+    assertThat(getNameOfDeclarer(routersAllOf.get(0)), is("scatterGather"));
+    assertThat(routersOneOf, hasSize(3));
+    assertThat(routersOneOf.stream().map(this::getNameOfDeclarer).collect(toList()),
+               containsInAnyOrder("choice", "firstSuccessful", "roundRobin"));
+    assertThat(scopesPassThrough, hasSize(3));
+    assertThat(scopesPassThrough.stream().map(this::getNameOfDeclarer).collect(toList()),
+               containsInAnyOrder("untilSuccessful", "parallelForeach", "try"));
+  }
+
+  private String getNameOfDeclarer(ParameterizedDeclarer declarer) {
+    return ((NamedDeclaration) declarer.getDeclaration()).getName();
+  }
+
+
+  private class MockConfigurer implements ComponentMetadataConfigurer {
```

Comment:
why not just use Mockito? you can use a captor to assert over the arguments of the configure() method

Reply:
The problem is that the `configure` method is separated from how it is configured, that's why I first store "which type it is" to then add it to the corresponding list

Reply:
you can still do that with mockito, but whatever is easier for you